### PR TITLE
Refactored Doors-and-levels game

### PR DESCRIPTION
**Pull request by Avyalikov**

### CODE REVIEW

File: HW2/Vyalikov_Anton/Doors-and-levels-0.2/DoorsAndLevels/PhraseProvider.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+using System.Text;
+using System.Threading.Tasks;
+using System.IO;
+using System.Xml;
+
+namespace DoorsAndLevels
+{
+    class PhraseProvider : Interfaces.IPhraseProvider
+    {
+        private Dictionary<string, string> keyValuePairs;
```
Comment (spore9) [inline HW2/Vyalikov_Anton/Doors-and-levels-0.2/DoorsAndLevels/PhraseProvider.cs]: Too general name, better to rename it to phrases or something like that

Comment (Avyalikov) [reply HW2/Vyalikov_Anton/Doors-and-levels-0.2/DoorsAndLevels/PhraseProvider.cs]: My mistake, thank you for comment.

File: HW2/Vyalikov_Anton/Doors-and-levels-0.2/DoorsAndLevels/Game.cs [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+                ioModule.WriteOutput(phraseProvider.GetMessage("FirstLevel"));
+            }
+            PrintList(doorsNumbers);
+        }
+
+        //Method is using for jump to the next level.
+        private void NextLevel(int door)
+        {
+            chosenDoors.Push(door);
+            prevDoors = doorsNumbers;
```
Comment (vsatkov) [inline HW2/Vyalikov_Anton/Doors-and-levels-0.2/DoorsAndLevels/Game.cs]: I have errors in numbers calculating after reach last level. I think this line is guilty.
When you use '=' for reference types you just copy references but not values. If at least one value has increased before break, we have wrong numbers on doors.

File: HW2/Vyalikov_Anton/Doors-and-levels-0.2/DoorsAndLevels/DoorsNumGenerator.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+            List<int> doorsNumbers = new List<int>();
+            Random genNum = new Random();
+            doorsNumbers.Add(0);
+
+            for (int i = 0; i < doorsCount - 1; i++)
+            {
+                while (true)
+                {
+
+                    int number = genNum.Next(2, 9);
```
Comment (vsatkov) [inline HW2/Vyalikov_Anton/Doors-and-levels-0.2/DoorsAndLevels/DoorsNumGenerator.cs]: You should use dependency of 'doorsCount' here. If I state 'DoorsCount in settings.xml is 10 I will get a loop.

File: HW2/Vyalikov_Anton/Doors-and-levels-0.2/DoorsAndLevels/PhraseProvider.cs [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+namespace DoorsAndLevels
+{
+    class PhraseProvider : Interfaces.IPhraseProvider
+    {
+        private Dictionary<string, string> phrasesDict;
+        private Dictionary<string, string> languages = new Dictionary<string, string>()
+        {
+            { "Rus", "..\\..\\..\\Res\\rusText.xml" },
+            {  "Eng", "..\\..\\..\\Res\\engText.xml" }
```
Comment (vsatkov) [inline HW2/Vyalikov_Anton/Doors-and-levels-0.2/DoorsAndLevels/PhraseProvider.cs]: You may change properties for json-file, like 'Copy to output directory' = 'Copy always' or another and will get simple request. But it's not necessary.

File: HW2/Vyalikov_Anton/Doors-and-levels-0.2/DoorsAndLevels/DoorsNumGenerator.cs [hunk 5]
```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace DoorsAndLevels
+{
+    class DoorsNumGenerator : Interfaces.IDoorsGenerator
+    {
+        public List<int> GetDoorsNumbers(int doorsCount, int minRandom, int maxRandom)
```
Comment (vsatkov) [inline HW2/Vyalikov_Anton/Doors-and-levels-0.2/DoorsAndLevels/DoorsNumGenerator.cs]: You didn't use minRandom and maxRandom in method, but it's good. Incorrect values may cause an error (see comment below).

Comment (Avyalikov) [reply HW2/Vyalikov_Anton/Doors-and-levels-0.2/DoorsAndLevels/DoorsNumGenerator.cs]: Stupid mistake, my fault, thank you

File: HW2/Vyalikov_Anton/Doors-and-levels-0.2/DoorsAndLevels/DoorsNumGenerator.cs (hunk 3 above)
Comment (Avyalikov) [reply HW2/Vyalikov_Anton/Doors-and-levels-0.2/DoorsAndLevels/DoorsNumGenerator.cs]: I didn't mind it, thank you

File: HW2/Vyalikov_Anton/Doors-and-levels-0.2/DoorsAndLevels/PhraseProvider.cs (hunk 4 above)
Comment (Avyalikov) [reply HW2/Vyalikov_Anton/Doors-and-levels-0.2/DoorsAndLevels/PhraseProvider.cs]: The thing is that I've already done it, and it didn't help, I don't know what's the problem.

**Pull request merged**